### PR TITLE
Separate dataset for encrypted volumes

### PR DESCRIPTION
**Pull request by giggsoff**
To support encrypted zvols we should move them to the child of persist/vault dataset to inherit options.

### CODE REVIEW

File: pkg/pillar/types/locationconsts.go [hunk 1]
```diff
@@ -25,8 +25,10 @@ const (
 	VolumeClearDirName = ClearDirName + "/volumes"
 	// PersistDebugDir - Location for service specific debug/traces
 	PersistDebugDir = PersistDir + "/agentdebug"
-	//VolumeZFSPool - pool for create volumes
-	VolumeZFSPool = "persist" + "/volumes"
+	//VolumeClearZFSDataset - dataset to create volumes without encryption
+	VolumeClearZFSDataset = "persist" + "/clear" + "/volumes"
+	//VolumeEncryptedZFSDataset - dataset to create volumes with encryption
+	VolumeEncryptedZFSDataset = "persist" + "/vault" + "/volumes"
```
Comment (rvs) [inline pkg/pillar/types/locationconsts.go:31]: This seems to pretty error prone -- I suggest you reconcile it with VolumeEncryptedDirName and VolumeClearDirName somehow

**Review by rvs — CHANGES_REQUESTED** [review]
LGTM aside from one small nit

File: pkg/pillar/zfs/zfs.go [hunk 2]
```diff
@@ -23,6 +23,27 @@ var (
...
+	for {
+		stdoutStderr, err = base.Exec(log, vault.ZfsPath, args...).CombinedOutput()
+		if err == nil {
+			return string(stdoutStderr), nil
+		}
+		tries++
+		if tries > maxTries {
+			break
+		}
+		time.Sleep(time.Second)
```
Comment (eriknordmark) [inline pkg/pillar/zfs/zfs.go:42]: Didn't you add a separate microservice a while back which would notify when the dataset is created? Or was that only for the zvol? 

Trying to understand if this is to handle the same thing that one can get an EINPROGRESS/EAGAIN and need to retry, and if so whether we have any confidence that under load it is sufficient with 3 tries in 3 seconds.

Comment (giggsoff) [reply pkg/pillar/zfs/zfs.go]: I removed this retry from here, it is not needed. Seems, we can hit errors with busy dataset on remove (DestroyDataset).

File: pkg/pillar/types/locationconsts.go (hunk 1 above)
Comment (giggsoff) [reply pkg/pillar/types/locationconsts.go]: I moved them to depend on *DirName.

**Review by zedi-pramodh — APPROVED** [review]
LGTM. Just for my understanding, this commit is just setting up the directories and actual encryption part will be in subsequent commit ?

**Comment by giggsoff** [comment]
> LGTM. Just for my understanding, this commit is just setting up the directories and actual encryption part will be in subsequent commit ?

Encryption is already here (persist/vault dataset), all what we need (and what is inside the PR) is to move our zvols to inherit options of existing dataset.

**Review by zededa-yuri — APPROVED** [review]
LGTM

**Review by eriknordmark — APPROVED** [review]
LGTM

**Review by rvs — APPROVED** [review]
LGTM

**Pull request merged**